cpp/oneapi/dal/algo/basic_statistics/backend/gpu/compute_kernel_dense_misc.hpp

```diff
@@ -0,0 +1,213 @@
+/*******************************************************************************
+* Copyright 2021 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#pragma once
+
+#include "oneapi/dal/backend/primitives/utils.hpp"
+#include "oneapi/dal/util/common.hpp"
+
+#ifdef ONEDAL_DATA_PARALLEL
+
+namespace oneapi::dal::basic_statistics::backend {
+
+namespace de = dal::detail;
+namespace be = dal::backend;
+namespace pr = dal::backend::primitives;
+
+enum class bs_list : std::uint64_t {
+    max = 1UL << 0,
+    min = 1UL << 1,
+    sum = 1UL << 2,
+    sum2 = 1UL << 3,
+    sum2cent = 1UL << 4,
+    mean = 1UL << 5,
+    sorm = 1UL << 6,
+    varc = 1UL << 7,
+    stdev = 1UL << 8,
+    vart = 1UL << 9
+};
+
+inline constexpr bs_list operator&(bs_list value_left, bs_list value_right) {
+    return bitwise_and(value_left, value_right);
+}
+
+inline constexpr bs_list operator|(bs_list value_left, bs_list value_right) {
+    return bitwise_or(value_left, value_right);
+}
+
+constexpr inline bs_list bs_mode_min_max = (bs_list::min | bs_list::max);
+constexpr inline bs_list bs_mode_mean_variance = (bs_list::mean | bs_list::varc);
+constexpr inline bs_list bs_mode_all =
+    (bs_list::min | bs_list::max | bs_list::sum | bs_list::sum2 | bs_list::sum2cent |
+     bs_list::mean | bs_list::sorm | bs_list::varc | bs_list::stdev | bs_list::vart);
+
+constexpr inline bs_list sum2cent_based_stat =
+    bs_list::sum2cent | bs_list::varc | bs_list::stdev | bs_list::vart;
+
+template <typename Float, bs_list List>
+class ndresult {
```

Comment:
Not a suitable name. It is not a N-dimension array, right?

Reply:
It is collection of 1-dimensional ndarrays, therefore named it with nd prefix

Reply:
what is your suggestion? Can suggest local_result local_buffer, is it ok?

Reply:
renamed as local_result and local_buffer_list